### ISSUE TITLE
saw-core-coq's `make` fails on master

### ISSUE THREAD

**Issue by m-yac** [GaloisInc/saw-core#174]
I first noticed this because the CI for this repo always fails.

Currently, `make` in `saw-core-coq/coq` fails with:
```
COQC generated/CryptolToCoq/SAWCorePrelude.v
File "./generated/CryptolToCoq/SAWCorePrelude.v", line 101, characters 51-52:
Error: Syntax error: '.' expected after [vernac:gallina] (in [vernac_aux]).
```
This can be fixed by just re-generating the scaffolding, i.e. running `saw generate_scaffolding.saw` in `saw-core-coq/saw`. However, we then we run into another problem:
```
COQC generated/CryptolToCoq/SAWCorePrelude.v
File "./generated/CryptolToCoq/SAWCorePrelude.v", line 835, characters 131-157:
Error:
In environment
a : Type
b : Type
c : Type
fn : (a -> CompM b) -> a -> CompM b
body : (a -> CompM b) -> CompM c
The term "TypesCons a b TypesNil" has type "InputOutputTypes"
while it is expected to have type "LetRecTypes".
```
This is because `InputOutputTypes` in `Prelude.sawcore` was replaced with `LetRecTypes` in both `Prelude.sawcore` and `saw-core-coq` on our `wip-heapster` branches, but when #128 was merged only the `saw-core-coq` changes were pulled into master. There are likely a few other instances of parallel changes to `saw-core` and `saw-core-coq` where only the latter got merged into master.

I would be happy to figure out what `saw-core` changes from `wip-heapster` are necessary for `saw-core-coq` to `make`  and put together a PR. However, since the goal is to move all heapster-specific Coq code out of this repo anyway, maybe that isn't worth it? Thoughts?

**Comment by eddywestbrook** [comment]
I think the goal of moving Heapster-specific Coq code out of this repo is a little longer term, and getting the Coq translation to work is more immediate, so it is definitely worth it to fix this stuff by merging any changes if necessary.

However, I do know that @ajayeeralla has been trying to get the Coq translation to work in master, however, so he has probably looked into this?

**Comment by ajayeeralla** [comment]
Unfortunately, I was getting those errors as well and couldn’t get time to look more in to them sorry!

**Comment by eddywestbrook** [comment]
Ah, ok, then it would definitely be worth it for @myac to fix this. If there were changes to `saw-core` that needed to be merged along with #128 in order to keep things working, then it definitely makes sense to merge those changes as well. (I assume you just mean changes to the prelude?)

**Comment by m-yac** [comment]
> Ah, ok, then it would definitely be worth it for @MYAC to fix this. If there were changes to saw-core that needed to be merged along with #128 in order to keep things working, then it definitely makes sense to merge those changes as well. (I assume you just mean changes to the prelude?)

Great I would be happy to put this together, and yes I assumed most of the changes would just be in the prelude.

**Comment by Ptival** [comment]
I agree with Eddy here, let's move in the wip-heapster code needed for the code to type-check, and we can then figure out the sweet spot for cutting what needs to stay in saw-core-coq and what can be Heapster library code.
